src/replicache.ts

```diff
@@ -926,6 +952,22 @@ export class Replicache<MD extends MutatorDefs = {}> {
     return {requestID, syncHead, ok: httpRequestInfo.httpStatusCode === 200};
   }
 
+  protected async _persist(): Promise<void> {
+    await this._ready;
+    const clientID = await this.clientID;
+    return this._persistLock.withLock(() =>
+      persist.persist(clientID, this._memdag, this._perdag),
+    );
+  }
+
+  private _schedulePersist(): void {
+    clearTimeout(this._schedulePersistID);
```

Comment:
if one is already scheduled, instead of delaying it further, maybe we just early return here?

```ts
if (this._schedulePersistID) {
  return;
}

this._schedulePersistID = window.setTimeout(() => {
  this._schedulePersistID = undefined;
  void this._persist();
}, PERSIST_TIMEOUT);
```

Reply:
Yeah, maybe. It is not clear what we want.

I think we do want to reschedule so that we do not interfere with more important work.

However, I think we might also want to ensure that persist is going to happen. Right now if there is a mutation every x ms and persist waits for y ms (y > x) we never call persist.

For now I think I'll go with every second and not reset the timer